### ISSUE TITLE
Error linking C examples on Linux

### ISSUE THREAD

**Issue by abhyshr** [GMLC-TDC/HELICS-src#345]
This issue is resolved now. Here's a summary of the error and its resolution

While compiling HELICS on ANL Linux workstations, I ran into an error on <b>linking the C examples</b>. The error was related to linking stdc++ library with the C examples.  Here's a short clip of what the error looked like.

```
../../src/helics/shared_api_library/libhelicsSharedLibd.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find(char const*, unsigned long) const@GLIBCXX_3.4.21'
../../src/helics/shared_api_library/libhelicsSharedLibd.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::append(char const*, unsigned long)@GLIBCXX_3.4.21'
/homes/abhyshr/soft/build_boost_1_65/lib/libboost_program_options.so.1.65.0: undefined reference to `VTT for std::__cxx11::basic_stringstream<char, std::char_traits<char>, std::allocator<char> >@GLIBCXX_3.4.21'
../../src/helics/shared_api_library/libhelicsSharedLibd.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string()@GLIBCXX_3.4.21'
```
This machine has three gcc versions; 4.8 (default), 5.1, and 6.2. HELICS needs gcc-4.9 and above so I was using gcc-6.2. I did this by updating the $PATH variable to pick-up gcc-6.2. However, what I did not do is to also set the LD_LIBRARY_PATH to pick up the corresponding libraries for gcc-6.2. Hence while compiling HELICS, stdc++ with gcc-4.8 was being linked, while the code was being compiled with gcc-6.2. The erroer was resolved on setting the LD_LIBRARY_PATH to point to the corresponding libraries for gcc-6.2

Long story short: Make sure you are linking with the correct gcc libraries (by setting the LD_LIBRARY_PATH correctly) when working on a system that has multiple compiler versions.

Update from @nightlark:
On the LC systems, they have a number of different compilers installed and use Environment Modules so that users can switch between them with a command like “module load gcc/4.9.3” – which looks like it takes care of updating environment variables so that the right libstdc++ gets found when compiling.

This is the best solution (if available). Check with your admin if they have environment module (or softenv system) to load the correct compilers, rather than fluxing with LD_LIBRARY_PATH yourself.
